conformance/tests/httproute-invalid-backendref-not-matching-listener-port.yaml

```
@@ -0,0 +1,28 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway-listener-not-matching-route-port
+  namespace: gateway-conformance-infra
+spec:
+  gatewayClassName: "{GATEWAY_CLASS_NAME}"
+  listeners:
+  - name: listener-1
+    port: 80
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: Same
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-listener-not-matching-route-port
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: gateway-listener-not-matching-route-port
+  rules:
+  - backendRefs:
+    - name: infra-backend-v1
+      kind: Service
+      port: 81
```

Maybe just a comment here to draw attention for future readers:
```suggestion
      # mismatched port here (81 is not an available gateway listener) triggers NoMatchingParent reason
      port: 81
```